ledger/participant-integration-api/src/main/scala/platform/index/IndexServiceImpl.scala

```diff
@@ -84,11 +104,182 @@ private[index] class IndexServiceImpl(
   ): Future[Option[ContractId]] =
     contractStore.lookupContractKey(readers, key)
 
+  // TODO DPP-1068: Copied from LfValueSerialization
+  private def apiIdentifierToDamlLfIdentifier(id: apiIdentifier): Ref.Identifier =
+    Ref.Identifier(
+      Ref.PackageId.assertFromString(id.packageId),
+      Ref.QualifiedName(
+        Ref.ModuleName.assertFromString(id.moduleName),
+        Ref.DottedName.assertFromString(id.entityName),
+      ),
+    )
+
+  private def computeInterfaceView(
+      templateId: Ref.Identifier,
+      record: com.daml.ledger.api.v1.value.Record,
+      interfaceId: Ref.Identifier,
+  )(implicit loggingContext: LoggingContext): com.daml.ledger.api.v1.event.InterfaceView = {
+    // TODO DPP-1068: The transaction stream contains protobuf-serialized transactions (Source[GetTransactionsResponse, NotUsed]),
+    //   we don't have access to the original Daml-LF value.
```

Comment:
how would you expect this to be improved? Since the value come from the db, I don’t see how we can avoid deserialization and the same for serialization afterwards. We can ofc slap a cache on this to solve the common case.

Reply:
We store the last 10k(?) transactions in memory, and transaction streams first try to get their data from that cache before going to the database ("in memory fan out"). Tailing transaction streams (those that are reasonably up to date with the ledger end) will get all their data from that cache.
Inside that cache, contract arguments are stored as `com.daml.lf.value.Value.VersionedValue`.

Reply:
makes sense, thanks for the explanation.